### PR TITLE
Component refinements 218

### PR DESCRIPTION
**Pull request by bgschiller**
Refer to #218 for requirements.

### CODE REVIEW

File: src/store/modules/geometry/actions/actions.js [hunk 1]
```diff
@@ -94,17 +24,25 @@ export default {
...
+  /*
+  * Erase the selection defined by a set of points on all faces on the current story
+  * Dispatched by the eraser tool
+  */
+  eraseSelection(context, payload) {
+    const
+      { points } = payload,
+      geometry_id = context.rootGetters['application/currentStoryGeometry'].id;
+
+    withPreservedComponents(context, geometry_id, () => {
```
Comment (bgschiller) [inline src/store/modules/geometry/actions/actions.js:36]: So happy with this code I wrote a blog post about it https://brianschiller.com/blog/2017/09/20/python-style-context-managers-in-js

**Comment by bgschiller** [comment]
I don't know why the /pr build is failing, but I think it's fine. ¯\\\_(ツ)_/¯

File: src/components/Grid/Grid.vue [hunk 2]
```diff
@@ -46,14 +58,21 @@ export default {
...
-        .yScale(v => this.rwuToGrid(v, 'y')),
+        .xScale(xScale)
+        .yScale(yScale),
+      drawDCGuideline: drawDaylightingControlGuideline()
+        .xScale(xScale)
+        .yScale(yScale),
     };
   },
   mounted() {
+    window.theGrid = this;
```
Comment (gh-23378) [inline src/components/Grid/Grid.vue]: something we might want to get rid of

File: src/components/Grid/methods.js [hunk 3]
```diff
@@ -7,9 +7,10 @@
 // THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER, THE UNITED STATES GOVERNMENT, OR ANY CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 const d3 = require('d3');
+window.d3 = d3;
```
Comment (gh-23378) [inline src/components/Grid/methods.js]: did we want d3 on the window?

File: test/e2e/specs/test.js [hunk 4]
```diff
@@ -85,4 +99,34 @@ module.exports = {
...
+      .perform(draw50By50Square)
+      .click('.tools [data-tool="Rectangle"]')
+      .click('#selections .add-sub-selection')
+      .perform(drawSquare(-10, 50, 10, 10))
+      .click('#selections .add-sub-selection')
+      .perform(drawSquare(-55, 40, 30, 20))
+      .execute(() => window.application.$store.state.geometry[0], [], ({ value }) => {
+        const { faces, edges } = value,
+          edgesConnect = (face) => {
+            console.log(`checking if edges connect for ${face.id}`);
```
Comment (gh-23378) [inline test/e2e/specs/test.js]: I know this is a test, but is this console.log something we want to keep?

**Pull request merged**